functions/source/rest_client.py

```diff
@@ -229,7 +248,15 @@ def create_workspaces(account_id, workspace_name, deployment_name, aws_region, c
     if customer_managed_key_id != '':
         DATA.update(MANAGEDKEYDATA)
 
-    response = post_request(URL, DATA, encodedbase64, version)
+    # Add deployment_name to the request object when provided, FYI Trial PAYG does not have a deployment_name or a deployment prefix
+    if deployment_name != '':
+        DATA.update(DEPLOYMENTDATA)
+
+    # Add customer_info to the request object when provided, for the OEM program
+    if customer_name != '':
```

Comment:
Don't we want to check all 3 attribute values are non-empty? i.e. customer_name, user_email and user_full_name?

Reply:
Additionally, you may want to add assertion in the template which ensures that either all the 3 values are provided or none, so that we alert the user in the console itself, rather than seeing a failure during API call.

Reply:
I added an assertion rule for all 3 OEM fields in the template so all 3 fields are mandatory. The rest_client.py does not have to be modified b/c if the customer_name is supplied then all 3 mandatory fields which only exists in the OEM template will be provided.